sql/src/test/java/io/crate/planner/TableStatsServiceTest.java

```diff
@@ -65,14 +69,79 @@ public void clearThreadPool() throws Exception {
     }
 
     @Test
+    public void testSettingsChanges() {
+        ClusterService clusterService = mock(ClusterService.class);
+
+        // Initially disabled
+        TableStatsService statsService = new TableStatsService(
+            Settings.builder().put(CrateSettings.STATS_SERVICE_REFRESH_INTERVAL.settingName(), 0).build(),
+            threadPool,
+            clusterService,
+            new NodeSettingsService(Settings.EMPTY),
+            () -> mock(SQLOperations.class));
+
+        assertThat(statsService.initialRefreshInterval(),
+            is(TimeValue.timeValueMinutes(0)));
+        assertThat(statsService.lastRefreshInterval(),
+            is(TimeValue.timeValueMinutes(0)));
+        assertThat(statsService.refreshScheduledTask(), is(nullValue()));
+
+        // Default setting
+        statsService = new TableStatsService(
+            Settings.EMPTY,
+            threadPool,
+            clusterService,
+            new NodeSettingsService(Settings.EMPTY),
+            () -> mock(SQLOperations.class));
+
+        assertThat(statsService.initialRefreshInterval(),
```

Comment:
imho this assertion can be removed, it's enough to check the current value like below